libraries/ESP8266WebServer/examples/FSBrowser/FSBrowser.ino

```diff
@@ -41,150 +77,180 @@ FS* filesystem = &LittleFS;
+    c = str.charAt(i);
+    if (c == '+') {
+      decodedString += ' ';
+    } else if (c == '%') {
+      i++;
+      code0 = str.charAt(i);
+      i++;
+      code1 = str.charAt(i);
+      c = (h2int(code0) << 4) | h2int(code1);
+      decodedString += c;
+    } else {
+      decodedString += c;
     }
-    DBG_OUTPUT_PORT.print("handleFileUpload Size: "); DBG_OUTPUT_PORT.println(upload.totalSize);
   }
+
+  return decodedString;
 }
 
-void handleFileDelete() {
-  if (server.args() == 0) {
-    return server.send(500, "text/plain", "BAD ARGS");
+
+/*
+   Checks filename for unsupported combinations
+   Returns an empty String if supported, or detail of error(s) if unsupported
+*/
+String getFileError(String path) {
+  String error = String();
+  if (!path.startsWith("/")) {
+    error += "!NO_LEADING_SLASH! ";
   }
-  String path = server.arg(0);
-  DBG_OUTPUT_PORT.println("handleFileDelete: " + path);
-  if (path == "/") {
-    return server.send(500, "text/plain", "BAD PATH");
+  if (path.indexOf("//") != -1) {
+    error += "!DOUBLE_SLASH! ";
   }
-  if (!filesystem->exists(path)) {
-    return server.send(404, "text/plain", "FileNotFound");
+  if (path.endsWith("/")) {
+    error += "!TRAILING_SLASH! ";
   }
-  filesystem->remove(path);
-  server.send(200, "text/plain", "");
-  path.clear();
+  return error;
 }
 
-void handleFileCreate() {
-  if (server.args() == 0) {
-    return server.send(500, "text/plain", "BAD ARGS");
-  }
-  String path = server.arg(0);
-  DBG_OUTPUT_PORT.println("handleFileCreate: " + path);
-  if (path == "/") {
-    return server.send(500, "text/plain", "BAD PATH");
-  }
-  if (filesystem->exists(path)) {
-    return server.send(500, "text/plain", "FILE EXISTS");
-  }
-  File file = filesystem->open(path, "w");
-  if (file) {
-    file.close();
+
+////////////////////////////////
+// Request handlers
+
+/*
+   Return the FS type, status and size info
+*/
+void handleStatus() {
+  DBG_OUTPUT_PORT.println("handleStatus");
+  FSInfo fs_info;
+  String json = String("{\"type\":\"") + fsName + "\", \"isOk\":";
+
+  if (fsOK) {
+    fileSystem->info(fs_info);
+    json += String("\"true\", \"totalBytes\":\"") + fs_info.totalBytes + "\", \"usedBytes\":\""  + fs_info.usedBytes + "\"";
   } else {
-    return server.send(500, "text/plain", "CREATE FAILED");
+    json += "\"false\"";
   }
-  server.send(200, "text/plain", "");
-  path.clear();
+  json += String(",\"unsupportedFiles\":\"") + unsupportedFiles + "\"";
+  json += "}";
+  server.send(200, "application/json", json);
 }
 
+
+/*
+   Return the list of files in the directory specified by the "dir" query string parameter.
+   Also demonstrates the use of chuncked responses.
+*/
 void handleFileList() {
+  if (!fsOK) {
+    return returnFail("FS INIT ERROR");
+  }
+
   if (!server.hasArg("dir")) {
-    server.send(500, "text/plain", "BAD ARGS");
-    return;
+    return returnFail("BAD ARGS");
+  }
+
+  String path = urlDecode(server.arg("dir"));
```

Comment:
Hmm. Unfortunately I have a very volatile memory...
All I can say is that I introduced server argument decoding a few weeks ago as part of this commit: https://github.com/esp8266/Arduino/pull/7182/commits/d709ae53049262c8dcf19eba9e5a40a98effca76#diff-d926380a25a7af1975d069c86e3d27d4 , and it was required to fix a bug with filenames containing blanks.
Now I admit after removing the URLdecode() I cannot reproduce the exact bug anymore :-(. 
It would be great to get rid of this code, but if I spent time finding how to URLdecode, I guess there was a good reason.
I'll have to dig further...

Reply:
It's quite possible. If you can't reproduce the bug, then I suggest removing this.

Reply:
In which case you no longer need the local urlDecode()?

Reply:
Yep, can't seem to be able to reproduce the bug.
Although I'm pretty sure I introduced it for a reason, the latest source works without it... And I went back to a 2.6.0 install and it still worked without it, so I don't know. Maybe I was chasing the wrong target when I introduced it. 
So indeed, no more call => no more urlDecode() implementation => no more hexDigitToInt(). I like that :-)